### PR TITLE
Added a break condition in case the installation failed on pack that …

### PR DESCRIPTION
**Pull request by hod-alpert**
…wasn't on the installation list

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
As seen in [this nightly build](https://app.circleci.com/pipelines/github/demisto/content/43378/workflows/2b1bf87f-bb05-474c-9c81-452aa28d8dac/jobs/187377) - There are two options to why it entered the endless loop.
1. The QRadar and CommonTypes packs are a mandatory dependencies for other packs in the list and that's why the installation failed with them even after they were removed.
2. The removal mechanism can only differ in one pack from the original packs to install that the method has received since the parameter `packs` is a set of packs made of `packs_to_install` - the malformed packs.

### CODE REVIEW

**Review by Itay4 — COMMENTED** [review]
1. lets trigger a build which simulates the scenario we are handling here against these changes to verify it works as expected
2. please add unit test

**Comment by hod-alpert** [comment]
@Itay4 Added unittests and verified it fails on current master (meaning it enters an endless loop)

[This](https://app.circleci.com/pipelines/github/demisto/content/43721/workflows/06d51a3b-44b3-420c-98af-4b699138bf98) is a nightly build that should pass with my changes only
And [this](https://app.circleci.com/pipelines/github/demisto/content/43719/workflows/8a6b7cfb-ac6e-4c49-8f4a-6f30e2c9e8af) is a nightly build that should show the pass after removing two different packs that are not existing in the production bucket.

File: Tests/Marketplace/Tests/search_and_install_packs_test.py [hunk 1]
```diff
@@ -245,3 +246,25 @@ def test_not_find_malformed_pack_id():
     with pytest.raises(Exception, match='The request to install packs has failed. '
                                         'Reason: This is an error message without pack ID'):
         script.find_malformed_pack_id('This is an error message without pack ID')
+
+@timeout_decorator.timeout(3)
+def test_install_nightly_packs_endless_loop(mocker):
```
Comment (Itay4) [inline Tests/Marketplace/Tests/search_and_install_packs_test.py:251]: please add test docstring

File: Tests/Marketplace/Tests/search_and_install_packs_test.py [hunk 2]
```diff
@@ -245,3 +246,25 @@ def test_not_find_malformed_pack_id():
     with pytest.raises(Exception, match='The request to install packs has failed. '
                                         'Reason: This is an error message without pack ID'):
         script.find_malformed_pack_id('This is an error message without pack ID')
+
+@timeout_decorator.timeout(3)
```
Comment (Itay4) [inline Tests/Marketplace/Tests/search_and_install_packs_test.py:250]: why is this needed?

Comment (hod-alpert) [reply Tests/Marketplace/Tests/search_and_install_packs_test.py:250]: Because theoretically this can lead to an endless loop

**Review by Itay4 — APPROVED** [review]
